### DISCUSSION TITLE
Can’t launch GUI of the terminal , error >psychometric class

### DISCUSSION THREAD

**Discussion by mnmneE** [wehr-lab/autopilot#106] [Q&A]
Hi there,

I followed the documentation related to the installation of autopilot; yet I am not able to launch the GUI of the terminal because error of python's binding of QT library that is Pyside2; I configured virtual environment and made sure that Python 3.7 is the main used version/ I wonder if is a bug  in the init file or if I did something wrong during the process!

please find attached screen shots about the terminal results/
in advance, thanks very much for your help.

Regards

![Captvirtver](https://user-images.githubusercontent.com/45367650/125451191-584237b7-7ed6-4173-b0d9-3d9d58167567.png)
![fcomm](https://user-images.githubusercontent.com/45367650/125451171-073bdac7-b262-4a5c-aade-f4fb1cb78074.png)
![com](https://user-images.githubusercontent.com/45367650/125451200-9dd971f2-3151-4104-a667-2862b96d9d69.png)

**Comment by sneakers-the-rat** [comment]
Yikes, thanks for bringing this up.

PySide moved a lot of its widget objects that were formerly part of `QtGui` into `QtWidgets`, this has been patched in the `dev` branch, which i am currently working on pulling back to `main`, but i have patched them separately in the meantime, try pulling!

https://github.com/wehr-lab/autopilot/pull/107

**Reply by mnmneE** [reply]
Thanks for your reply and help, actually I tried again and I couldn't launch the terminal because of an error also related to Qt library ; please find attached a screenshot for the reported error; in advance thanks for your help
![Screenshot from 2021-07-21 17-29-36](https://user-images.githubusercontent.com/45367650/126516586-deda1f42-e80f-4021-b92c-637d28dd347c.png)

**Reply by sneakers-the-rat** [reply]
ah yes, this is a reasonably common problem with Qt5. 

so the error:
```
Could not load the Qt platform plugin "xcb" in "" even though it was found
```

This doesn't show up on the CI (which installs and launches the GUI on a fresh linux image) and that might be pretty telling. See this update i made to the travis config in the dev branch:

https://github.com/wehr-lab/autopilot/blob/8c078ee96fafb324770cb71035da667dccc24348/.travis.yml#L17-L28

- qt depends on libxcb https://doc.qt.io/qt-5/linux-requirements.html
- and here are the ubuntu libxcb related libraries: https://packages.ubuntu.com/source/bionic/libxcb
- it looks like the napari folks had a similar issue https://github.com/napari/napari/issues/1457 , specifically see here https://github.com/napari/napari/issues/1457#issuecomment-752667286

I think you should try installing/reinstalling xcb. I'm not sure what the absolute minimal combination of packages here, but just taking inspiration from the travis config which works:

```
sudo apt update
sudo apt install --reinstall -y  \
  libxcb-icccm4 \
  libxcb-image0 \
  libxcb-keysyms1 \
  libxcb-randr0 \
  libxcb-render-util0 \
  libxcb-xinerama0 \
  libxcb-xfixes0 
```

i've opened an issue https://github.com/wehr-lab/autopilot/issues/108 to remind myself to add those in to 
- the scripts (we had managed to get away with the terminal not needing any system packages until now, but can't stay lucky forever :p) like that for the pilot agents: https://github.com/wehr-lab/autopilot/blob/30afe22720e1fd6676c8ff90569f6e34ba6faec0/autopilot/setup/scripts.py#L39-L46
- and the installation docs

lmk if that doesn't work :)

**Reply by mnmneE** [reply]
Thanks alot for this comprehensive reply; I will try to re/install the xcb and may check if there are other commands to do for it, then will keep you updated.

**Reply by mnmneE** [reply]
Hi, after installing xcb , the terminal GUI is successfully launched; thanks very much for your help.

**Reply by sneakers-the-rat** [reply]
excellent! i'll be pulling in a lot of work in the next few days, nothing should be too breaking, but no need to be shy with any further issues you will inevitably run into :)
